HLTrigger/JetMET/plugins/HLTScoutingPFProducer.cc

```diff
@@ -151,7 +155,17 @@ void HLTScoutingPFProducer::produce(edm::StreamID sid, edm::Event &iEvent, edm::
             break;
           ++index_counter;
         }
-        outPFCandidates->emplace_back(cand.pt(), cand.eta(), cand.phi(), cand.mass(), cand.pdgId(), vertex_index);
+
+        if (storeWithHalfPrecisionFloatingPoint) {
+          outPFCandidates->emplace_back(MiniFloatConverter::float16to32(MiniFloatConverter::float32to16(cand.pt())),
+                                        MiniFloatConverter::float16to32(MiniFloatConverter::float32to16(cand.eta())),
+                                        MiniFloatConverter::float16to32(MiniFloatConverter::float32to16(cand.phi())),
+                                        MiniFloatConverter::float16to32(MiniFloatConverter::float32to16(cand.mass())),
```

Comment:
is there no way to just truncate the precision, without converting to 16 bits and back to 32 bits ?

Reply:
of course there is: it is called bfloat16 (instead of halffloat)
https://en.wikipedia.org/wiki/Bfloat16_floating-point_format

BUT it really has low precision

bfloat16 is what it is mostly used for AI.

On Intel machines (AVX2 and more) one could use intrinsics to perform the conversion to/from halffloat (worth only if contiguous as in a SOA)

Reply:
Mhm,  no, that doesn't seem like a good candidate here.

I just meant to ask if there is way to do `MiniFloatConverter::float16to32(MiniFloatConverter::float32to16(x))` in a single call.

Reply:
under the assumption that the first call did not over/underflow it is trivial yes
(well, one must properly round, not truncate. still pretty trivial).

but at this point I do not understand what is going on, I was thinking data were stored in hf16 (aka int16), not in truncated float32
(ok I missed "let the compression reduce the information being stored.")

Reply:
> but at this point I do not understand what is going on, I was thinking data were stored in hf16 (aka int16), not in truncated float32

that was my assumption as well

on the other hand, storing the truncated precision back in a float32 has the advantage of not affecting the data format, and apparently the compression is still good enough to reduce the size by ~37%